File: app/src/main/java/za/org/grassroot2/model/RequestMapper.java
package za.org.grassroot2.model;

import java.util.ArrayList;
import java.util.List;

import za.org.grassroot2.model.contact.Contact;
import za.org.grassroot2.model.request.MemberRequest;
import za.org.grassroot2.model.util.PhoneNumberFormatter;


public class RequestMapper {
    private static MemberRequest map(Contact c) {
        MemberRequest result = new MemberRequest();
        result.displayName = c.getDisplayName();
        result.emailAddress = c.getEmailAddresses().isEmpty() ? null : c.getEmailAddresses().get(0);
        result.alternateNumbers.addAll(c.getPhoneNumbers());
        result.phoneNumber = PhoneNumberFormatter.formatNumberToE164(c.getPhoneNumbers().get(0));
        result.createdDate = System.currentTimeMillis();
        return result;
    }

    public static List<MemberRequest> map(String groupUid, List<Contact> contacts) {
        List<MemberRequest> body = new ArrayList<>();
        if(contacts != null){
            for (Contact c : contacts) {
                MemberRequest request = map(c);
                request.groupUid = groupUid;
                body.add(request);
            }
        }
        return body;
    }
}

File: app/src/main/java/za/org/grassroot2/model/RequestMapper.kt
package za.org.grassroot2.model

import java.util.ArrayList

import za.org.grassroot2.model.contact.Contact
import za.org.grassroot2.model.request.MemberRequest
import za.org.grassroot2.model.util.PhoneNumberFormatter


object RequestMapper {
    private fun map(c: Contact): MemberRequest {
        val result = MemberRequest()
        result.displayName = c.displayName
        result.emailAddress = if (c.emailAddresses.isEmpty()) null else c.emailAddresses[0]
        result.alternateNumbers.addAll(c.phoneNumbers)
        result.phoneNumber = PhoneNumberFormatter.formatNumberToE164(c.phoneNumbers[0])
        result.createdDate = System.currentTimeMillis()
        return result
    }

    fun map(groupUid: String, contacts: List<Contact>?): List<MemberRequest> {
        val body = ArrayList<MemberRequest>()
        if (contacts != null) {
            for (c in contacts) {
                val request = map(c)
                request.groupUid = groupUid
                body.add(request)
            }
        }
        return body
    }
}
